tools/python/clean_strings_txt.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python2.7
+
+from __future__ import print_function
+
+import logging
+import re
+import subprocess
+from argparse import ArgumentParser
+from collections import defaultdict
+from itertools import chain
+from os.path import abspath, isabs
+
+from find_untranslated_strings import StringsTxt
+
+"""
+This script determines which strings are used in the platform code (iOS and
+Android) and removes all the other strings from strings.txt. For more information,
+run this script with the -h option.
+"""
+
+
+OMIM_ROOT = ""
+
+MACRO_RE =  re.compile('L\(.*?@\"(.*?)\"\)')
+XML_RE = re.compile("value=\"(.*?)\"")
+ANDROID_JAVA_RE = re.compile("R\.string\.([\w_]*)")
+ANDROID_XML_RE = re.compile("@string/(.*?)\W")
+IOS_CANDIDATES_RE = re.compile("(.*?):[^L\(]@\"([a-z0-9_]*?)\"")
+
+HARDCODED_CATEGORIES = None
+
+HARDCODED_COLORS = [
+    "red", "yellow", "blue", "green", "purple", "orange", "brown", "pink"
+]
+
+
+def exec_shell(test, flags):
```

Comment:
Much better, as you don't need to call exec_shell with "" as a second argument now, also you don't need to join multiple arguments to a single one.